src/main/kotlin/hr/foi/pop/backend/controllers/AuthenticationController.kt

```diff
@@ -0,0 +1,119 @@
+package hr.foi.pop.backend.controllers
+
+import hr.foi.pop.backend.definitions.ApplicationErrorType
+import hr.foi.pop.backend.exceptions.*
+import hr.foi.pop.backend.models.user.User
+import hr.foi.pop.backend.models.user.UserLoginResponseDTO
+import hr.foi.pop.backend.models.user.UserMapper
+import hr.foi.pop.backend.request_bodies.LoginRequestBody
+import hr.foi.pop.backend.request_bodies.RefreshTokenRequestBody
+import hr.foi.pop.backend.request_bodies.RegisterRequestBody
+import hr.foi.pop.backend.responses.ErrorResponse
+import hr.foi.pop.backend.responses.SuccessResponse
+import hr.foi.pop.backend.responses.WarningResponse
+import hr.foi.pop.backend.security.jwt.TokenPair
+import hr.foi.pop.backend.services.AuthenticationService
+import hr.foi.pop.backend.services.UserService
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+
+@RestController
+@RequestMapping("/api/v2/auth")
+class AuthenticationController {
+    @Autowired
+    lateinit var userService: UserService
+
+    @Autowired
+    lateinit var authenticationService: AuthenticationService
+
+    @PostMapping("register")
+    fun registerUser(@RequestBody request: RegisterRequestBody): ResponseEntity<SuccessResponse> {
```

Comment:
Dear @mislavmatijevic,

I am sorry for not finding this earlier, but I think you should consider changing email validator, because:
![image](https://github.com/foivz/pop-app-backend/assets/61520141/17455d51-f60d-44a2-bbce-23c3ee4d9d2a)

Reply:
Fixed via [d03bea](https://github.com/foivz/pop-app-backend/pull/22/commits/d03beaf28e89ab7d2aa97dc321437d721bc57790).